quickwit/quickwit-query/src/elastic_query_dsl/match_bool_prefix.rs

```diff
@@ -40,7 +40,7 @@ impl ConvertableToQueryAst for MatchBoolPrefixQuery {
             tokenizer: None,
             mode: crate::query_ast::FullTextMode::BoolPrefix {
                 operator: self.params.operator,
-                max_expansions: default_max_expansions(),
+                max_expansions: u32::MAX,
```

Comment:
Is it in line with the default behavior in ES?

Reply:
as far as I can tell, `match_bool_prefix` has a default max_expansions of 50, but does not honor it. For instance if I add 52 documents with a term from `xaa` to `xbz`, I would expect only 50 documents to be returned in a `match_bool_prefix` for `x`, but they all are (in fact, they all are even if I add `"max_expansions": 1` to the query

Reply:
can you add a comment about that so that the next reader won't be lost?